examples/pygfx-example.py

```diff
@@ -0,0 +1,209 @@
+"""
+Followed the following guide:
+https://pomb.us/build-your-own-react/
+and implemented it in Python.
+
+Example 'React' code:
+
+<Group name="Landmarks">
+  <Point name="Hip" position="[1, 1, 1]" />
+</Group>
+
+This can be transpiled to a tree of 'VNodes'.
+Instead of JSX/Vue template, this could also be just creating
+the tree of VNodes 'manually'.
+VNodes are essentially a description of the to-be-displayed item.
+
+1. Create tree of VNodes
+2. Render the tree (see __init__.py)
+    - Split up the work in smaller chunks (node-by-node)
+    - Build up a 'work-in-progress' tree of 'fibers'
+    - Once it is ready, compare this tree with the 'current' tree (reconciliation)
+    - The renderer that is called during reconciliation creates the actual 'dom' nodes,
+      so a PygfxRenderer for instance would know how to reconcile a tree of pygfx
+      objects.
+
+"""
+import logging
+import random
+
+import pygfx as gfx
+from wgpu.gui.auto import run, WgpuCanvas
+
+from pygui import create_element as h, EventLoopType, PyGui
+from pygui.renderers import PygfxRenderer
+
+try:
+    # If rich is available, use it to improve (traceback) logs
+    from rich.logging import RichHandler
+    from rich.traceback import install
+    import shutil
+
+    terminal_width = shutil.get_terminal_size((100, 20)).columns - 2
+    install(width=terminal_width)
+
+    FORMAT = "%(message)s"
+    logging.basicConfig(
+        level="WARN",
+        format=FORMAT,
+        datefmt="[%X]",
+        handlers=[RichHandler(rich_tracebacks=True)],
+    )
+except ModuleNotFoundError:
+    pass
+
+
+logger = logging.getLogger(__name__)
+
+
+class SelectableObjectsCanvas(WgpuCanvas):
+    def handle_event(self, event):
+        if event["event_type"] == "pointer_down":
+            info = renderer.get_pick_info((event["x"], event["y"]))
+            wobject = info["world_object"]
+            if wobject:
+                if handle_event := getattr(wobject, "handle_event", None):
+                    handle_event({"event_type": "click"})
+
+        super().handle_event(event)
+
+
+sphere_geom = gfx.sphere_geometry(radius=0.5)
+materials = {
+    False: gfx.MeshPhongMaterial(color=[1, 1, 1]),
+    True: gfx.MeshPhongMaterial(color=[1, 0, 0]),
+}
+
+
+def PointCloud(props):
+    random.seed(0)
+
+    # Set some default values for props
+    props.setdefault("selected", -1)
+    props.setdefault("count", 50)
+
+    def set_selected(index):
+        logger.debug(f"select: {index}")
+        if props["selected"] == index:
+            props["selected"] = -1
+        else:
+            props["selected"] = index
+
+    def random_point(index, selected):
+        return h(
+            "Mesh",
+            {
+                "geometry": sphere_geom,
+                "material": materials[index == selected],
```

Comment:
Just curious: I'm guessing observ just leaves these classes alone and unreactive right?

Reply:
I believe so. I don't know if the example covers such a scenario, but a test for this seems like a great addition for the test suite of observ :)

Reply:
See https://github.com/fork-tongue/observ/issues/67